src/utils/callbackSet.ts

```diff
@@ -0,0 +1,38 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+  export module Utils {
+    export class CallbackSet<CB extends Function> {
+      private _values: CB[];
+
+      constructor() {
+        this._values = [];
+      }
+
+      public add(value: CB) {
+        if (this._values.indexOf(value) === -1) {
```

Comment:
Agree with @bluong as there is an inherent assumption for any operations on a `Set`.